live_site.py

```diff
@@ -0,0 +1,433 @@
+	for domain, details in enumerate(my_domains['domains']):
+		for key, value in details.items():
+			x = details['name']            
+			if domain_name != x: continue
+			domain_exists = x
+			return domain_exists
+		else:
+			return None	
+	else:
+		return None
+
+
+def will_add_domain(droplet, domain_name):
+		# IP address is needed to create the domain name, so we call the function get_hostname_ip()
+		ip = get_hostname_ip(droplet)
+		# Just want to make sure the ip_address is a string, so I convert it anyway.
+		ip_address = str(ip)  
+
+		# This adds/creates the domain name
+		new_domain = add_new_domain(domain_name, ip_address)
+
+		if new_domain is not None:
+			click.secho('>>> Your new domain name {} has been created with IP Address {}. '.format(domain_name, ip_address), fg='bright_green')
+		else:
+			click.secho('[!] Request Failed', fg='bright_red')
+
+
+def create_domain(droplet, domain_name):
+
+	domain_exists = check_domain(domain_name)
+	click.echo(domain_exists)
+
+	if domain_exists == domain_name:
+		click.secho('>>> Your domain name {} already exists!'.format(domain_name), fg='bright_yellow')
+		click.secho('>>> We are now removing it to droplet IP mapping ...', fg='bright_yellow')
+		del_domain(domain_name)
+		click.secho('Creating a new domain name now...', fg='bright_yellow')
+		will_add_domain(droplet, domain_name)
+	else:
+		click.echo('...')
+		will_add_domain(droplet, domain_name)
+
+
+def add_new_domain_record(type, name, data, ttl, domain_name):
+
+	api_url = '{0}domains/{1}/records'.format(api_url_base,domain_name)    
+
+	records = {"type":type,"name":name,"data":data,"ttl":ttl}
+
+	response = requests.post(api_url, headers=headers, json=records) 
+
+	if response.status_code == 201:
+		records = json.loads(response.content)
+		return records
+	else:
+		click.secho('[?] Unexpected Error: [HTTP {0}]: Content: {1}'.format(response.status_code, response.content), fg='bright_red')
+		return None                     
+		
+
+def add_domain_rec(droplet, domain_name):
+	# Get the IP Address associated with jediluke.space
+	ip = get_hostname_ip(droplet)
+	rec_data = str(ip)
+
+	add_response = add_new_domain_record('A','www',rec_data,1800, domain_name)
+
+	if add_response is not None:
+		click.secho('>>> Your A Record has been created... ', fg='bright_green')
+		for k, v in add_response.items():
+			click.secho('  {0}:{1}'.format(k, v), fg='bright_green')
+	else:
+		click.secho('[!] Request Failed', fg='bright_red')
+
+def set_nginx(droplet, domain_name):
+
+	host = get_hostname_ip(droplet)
+
+	client = SSHClient(host, user='root', password='', pkey=ssh_key,port=22)
+	click.secho('Connecting to your server...', fg='bright_yellow')
+
+	def upload():
+	# This funtion will:
+	# Upload the initial files and configuration needed in setting up Nginx to the server   
+
+		click.echo('Setting up your configuration block for {}...'.format(domain_name))
+		loc_file1 = 'D:/Workbench/Luke/jediluke.space/files/jediluke.space.txt'
```

Comment:
One thing we probably want to do here is use relative file paths

This article looks like a good read about file paths https://phpdelusions.net/articles/paths

If I go to run this python on my computer, well for starters, I don't even have a D drive :D and I have put this project in a totally different spot to you, so this will fail.

BUT, that file `jediluke.space.txt` is very close to the root directoy of the project (which is what I copied down when I cloned the project using git). So If I was in the root directory of the project, then relative to the **working directory** (the "place" or directory I am in my shell when I'm running this script), it's just in the `files/jediluke.space`

So can you read that article, and once you understand why, change this to something like './files/jediluke.space.txt'